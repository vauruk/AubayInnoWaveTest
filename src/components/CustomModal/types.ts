import {StyleProp, ViewStyle} from 'react-native';

export interface Props {
  testID?: string;
  title?: any;
  // style?: StyleProp<ViewStyle>;
  children?: any;
  isVisible: boolean;
  textConfirm?: string;
  textCancel?: string;
  type?: 'error' | 'warning' | 'success' | 'info';
  handleCancel?: () => void;
  handleSuccess?: () => void;
}
