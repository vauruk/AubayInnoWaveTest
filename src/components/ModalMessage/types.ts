import {StyleProp, ViewStyle} from 'react-native';

export interface Props {
  testID?: string;
  title?: any;
  // style?: StyleProp<ViewStyle>;
  children?: any;
  isVisible: boolean;
  textConfirm?: string;
  textCancel?: string;
  type?: 'error' | 'warning' | 'success';
  handleCancel?: () => void;
  handleSuccess?: () => void;
}
