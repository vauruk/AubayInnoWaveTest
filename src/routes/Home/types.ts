export enum HomeRoutes {
  index = '/home',
  HomeLogged = '/home/logged',
  EditDevice = '/home/EditDevice',
  Splash = '/home/Splash',
}

export type HomeStackParamsList = {
  [HomeRoutes.HomeLogged]: undefined;
  [HomeRoutes.index]: undefined;
  [HomeRoutes.Splash]: undefined;
};
