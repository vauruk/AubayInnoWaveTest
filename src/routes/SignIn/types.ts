export enum SignInRoutes {
  index = '/signin',
  Splash = '/signin/splash',
}

export type SignInStackParamsList = {
  [SignInRoutes.Splash]: undefined;
};
