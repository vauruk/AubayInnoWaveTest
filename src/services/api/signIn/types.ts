/**
 * {
    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJtaWEuY2lhbGF6ZXIuY29tLmJyIiwic2NvcGUiOiJ0b2tlbjphYzp1c2VyOmltcCIsImF1ZCI6ImFwcGNpYSIsImlhdCI6MTY1ODQ1Mjg2NSwiZXhwIjoxNjU5NzQ4ODY1LCJzdWIiOiJNdz09Iiwicm9sZSI6Im93bmVyIn0.sVnhAsvEA5Et6Ghmb599KNOaML-26posQ5mKAuUHRNjKZFc2hKMpYF4S20tDaOqLHZB7_gPjum_UTg9tI5v8dKafAhgad8TFX-xWyNjkgfmyVBJHlQZayn_AxshkssqNFhyKaa1RD4sP3WY-KiB-aOZcW191nANFxOp5tGoJ8PSk5Dj1IAd0w-Vya-a0y99i-KuEmb82IqfdTVx17qDOQxzjo4UlbbBSoFthazXXPpbAri7jFqH1vAqNuqyMnsXFkrG2fwUDwkCZ5LduSW80YKoVrG7iOcbc3Gmj9wqndi6BY2K8f9R9VQn275p4izaJyOspOtqSsoKuUfcDxplSlIohSGreWyaCtSpmTToEf7QPgeO5V345zNR3GQyGCzqtWdeqeObEmgn-CfIHHUvRIR8wna5Ukjhi-j-At38qRQWZ_vjjh5VIm3x_bFi8oAXhdPu0D_kpHJUEDVomdSiMP6FfI3Ay6Ee5m_oqgda-fHjr_TDTp6jqI9q4HqV1xDyN1KDMz1Rk7HSR8ypP23vkANm36JQWCDXSyW1Y0Vmmvwqdj9Rb_30Ps_qeIlyK8CvDHdtJgqCo3dXdQoxrfHwPa96IHImKe7im_k6lHqlDd8rRV7rLsfbohPGIalOgiIfDQ_W5e60IMH_EKJ6m6GyqMGwbnx-7z2bmJuMFRLGHOzQ",
    "token_type": "Bearer",
    "expires": 1659748865,
    "role": "owner"
}
*/

export interface IAuthResponse {
  access_token: string;
  token_type: string;
  expires: number;
  role: string;
}

export interface IAuthData {
  user: string | number | undefined;
  password: string | number | undefined;
}
