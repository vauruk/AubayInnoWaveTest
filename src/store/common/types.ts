// export interface Field {
//   value: string | number | undefined;
//   isValid?: boolean;
//   error?: string | undefined;
// }

// export const initialFieldState: Field = {
//   value: undefined,
//   error: undefined,
//   isValid: false,
// };
