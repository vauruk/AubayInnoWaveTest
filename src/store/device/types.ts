/**
 * @author Vanderson de Moura Vauruk
 *
 */
import {IDevice} from '../../views/EditDevice/types';

export interface SetField {
  fieldName: FieldLabel;
  value: string | number | undefined;
}

export interface UserLoggedResponse {
  token: string | undefined;
  access_token: string;
  token_type: string;
  expires: number;
  role: string;
}

export interface FormState {
  token: string | undefined;
  loading: boolean;
  submitError: string | undefined;
  deviceData: IDevice | undefined;
  listDevice: Array<IDevice>;
}

export enum FieldLabel {
  model = 'model',
  os = 'os',
  currentOwner = 'currentOwner',
  notes = 'notes',
}
