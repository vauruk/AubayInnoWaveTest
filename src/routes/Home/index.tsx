import React from 'react';
import {HomeStackParamsList} from './types';
import {routes} from './config';
import {createStackNavigator} from '@react-navigation/stack';

const Stack = createStackNavigator<HomeStackParamsList>();

const HomeNavigator = () => (
  <Stack.Navigator>
    {routes.map(route => (
      <Stack.Screen
        key={route.name}
        name={route.name as any}
        options={route.options}>
        {route.component}
      </Stack.Screen>
    ))}
  </Stack.Navigator>
);

export default HomeNavigator;
