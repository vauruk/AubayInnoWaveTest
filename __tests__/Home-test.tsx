/**
 * @format
 */

import 'react-native';
import React from 'react';
import Home from '../src/views/Home';

import {
  render,
  screen,
  fireEvent,
  act,
  cleanup,
} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';
import {rootStore} from '../src/store';
import {ThemeProvider} from '@rneui/themed';
import {theme} from '../src/theme';
import {createStackNavigator} from '@react-navigation/stack';
import {HomeRoutes} from '../src/routes/Home/types';
import {EditDevice} from '../src/views';
import HeaderButton from '../src/views/Home/headerButton';

afterEach(cleanup);
const RootStack = createStackNavigator();
const MockedNavigator = ({component, params = {}}) => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={rootStore}>
        <NavigationContainer>
          <RootStack.Navigator>
            <RootStack.Screen
              name="MockedScreen"
              component={component}
              initialParams={params}
            />
          </RootStack.Navigator>
        </NavigationContainer>
      </Provider>
    </ThemeProvider>
  );
};
const MockedNavigatorForRoot = ({route, params = {}}) => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={rootStore}>
        <NavigationContainer>
          <RootStack.Navigator initialRouteName={route}>
            <RootStack.Screen
              name={HomeRoutes.HomeLogged}
              component={Home}
              initialParams={{headerLeft: () => <HeaderButton />}}
            />
            <RootStack.Screen
              name={HomeRoutes.EditDevice}
              component={EditDevice}
              initialParams={params}
            />
          </RootStack.Navigator>
        </NavigationContainer>
      </Provider>
    </ThemeProvider>
  );
};

test('Open Home Screen', async () => {
  const rendered = render(<MockedNavigator component={Home} />).toJSON;
  expect(rendered).toMatchSnapshot();

  const listDevice = await screen.findByTestId('flatList');
  expect(listDevice).toBeTruthy();
  // Add this line to get rid of the act() warning
  // await waitFor(() => {});
});

describe('Making test Devices Liste Edit', () => {
  it('should match the snapshot EditDevice', () => {
    const rendered = render(<MockedNavigator component={EditDevice} />).toJSON;

    expect(rendered).toMatchSnapshot();
  });

  it('should match the snapshot Home', () => {
    const rendered = render(<MockedNavigator component={Home} />).toJSON;

    expect(rendered).toMatchSnapshot();
  });

  it('test imputText Edit Device', async () => {
    const expectedValueOs = 'android';
    render(<MockedNavigator component={EditDevice} />);

    const inputModelId = screen.getByTestId('modelId');
    await act(async () => {
      await fireEvent(inputModelId, 'onChangeText', expectedValueOs);
    });
    const next = screen.queryByText('modelId');

    expect(next).toEqual(next);
  });
  test('clicking on one item takes you to the details screen', async () => {
    render(<MockedNavigatorForRoot route={HomeRoutes.EditDevice} />);
    const input = await screen.getByTestId('modelId');
    const model = 'Samsung';
    fireEvent.changeText(input, model);
    console.log(input);
    const next = screen.queryByText(model);

    expect(next).toEqual(next);
    fireEvent.press(screen.getByText('Save'));
  });
});
