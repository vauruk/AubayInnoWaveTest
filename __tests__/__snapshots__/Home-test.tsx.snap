// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Making test Devices Liste Edit should match the snapshot EditDevice 1`] = `[Function]`;

exports[`Making test Devices Liste Edit should match the snapshot Home 1`] = `[Function]`;

exports[`Open Home Screen 1`] = `[Function]`;
